bm/bm_emit.cpp

```diff
@@ -100,7 +100,7 @@ void bm_fyaml_ostream(bm::State& st)
     c4::csubstr src = c4::to_csubstr(s_bm_case->src);
     std::ostringstream os;
     std::string str;
-    fy_emitter_cfg cfg = {};
+    fy_emitter_cfg cfg;
```

Comment:
does g++4.8 do the right thing here? Will this be zero-initialized? I'm not sure.

Reply:
I don't think so, but the structure is initialized below

Reply:
I'm now having the same issue on incoming modifications (TagDirective in tree.cpp) which is the reason why CI broke.

I was trying to fix the following compiler warning (just saw that it's only a warning):
```
error: missing initializer for member 'c4::yml::TagDirective::handle' [-Werror=missing-field-initializers]
         m_tag_directives[i] = {};
                             ^
```

I can try to suppress that one for gcc-4.8, I'll run a couple tests to see if zero init is properly achieved.

An alternative is to default-init the values in class (which should act like providing a proper constructor):
```
struct TagDirective
{
    csubstr handle;
    csubstr prefix;
    size_t next_node_id = 0;
};
```
But drawback seems to be that then thing such as `m_tag_directives[i] = {};` becomes invalid (since a constructor is implicitly provided), not sure why C++ became so awkward on initialization.

Reply:
Yes, I think some of the changes in c++11 were too hasty, and things have compounded since then. Initialization C++ nowadays is a beast best left quiet. So be sure the changes you do there do not have side effects.